### ISSUE TITLE
Unable to post ad

### ISSUE THREAD

**Issue by Rings4** [ArthurG/Kijiji-Repost-Headless#182]
I'm unable to post an ad. It gets to uploading all the images, but then fails at the very end. The html file it generates is like an incomplete version of the upload page with my 5 images that says "Please fix the errors on the page" at the top. 

Here's a screenshot of the cmd window when this occurs: https://i.imgur.com/tmSf3vJ.png

What am I missing? I've already made sure the requirements.txt's contents are all installed. Been using this for months until the log in bug that only recently got fixed, not sure what else could be wrong. I re-created the *.yaml file and I still get this error.

**Comment by michaelkuzmin** [comment]
there is something wrong with your yml file. most likely one of the parameters is missing. how you fix this is you post an ad manually and look at the Chrome - Developer Tools - Network. you will find all the parameters there when you look at the requests to Kijiji. compare with your YML file. some categories are more tricky than others.

**Comment by Rings4** [comment]
> there is something wrong with your yml file. most likely one of the parameters is missing. how you fix this is you post an ad manually and look at the Chrome - Developer Tools - Network. you will find all the parameters there when you look at the requests to Kijiji. compare with your YML file. some categories are more tricky than others.

Thanks, I'll look into it when I get on. If that is the case, I guess the issue here then would be the fact that the tool cannot create proper yml files anymore.

**Comment by Rings4** [comment]
> 
> 
> there is something wrong with your yml file. most likely one of the parameters is missing. how you fix this is you post an ad manually and look at the Chrome - Developer Tools - Network. you will find all the parameters there when you look at the requests to Kijiji. compare with your YML file. some categories are more tricky than others.

I'm trying this on FireFox (F12 - Network tab) but I don't see anything similar to the fields in the yml. Can you elaborate on how I can find the fields according to the yml please?

**Comment by Rings4** [comment]
Just tried the new version which was updated 7 days ago. Still doesn't work. It's been over a month, would be great to see this fixed!

**Comment by jackm** [comment]
What specific category are you trying to post to? Perhaps I can find the extra parameters you need.

It is a known problem that this tool doesn't always include every ad parameter that Kijiji requires for certain categories. If they are missing, the Kijiji site will reject the ad and not post it.

**Comment by Rings4** [comment]
> 
> 
> What specific category are you trying to post to? Perhaps I can find the extra parameters you need.
> 
> It is a known problem that this tool doesn't always include every ad parameter that Kijiji requires for certain categories. If they are missing, the Kijiji site will reject the ad and not post it.

The category ID my yaml file has is 14922002. 
I'm trying to post in Buy & SellAudioStereo Systems & Home Theatre

What's strange is that even the old yml and yaml files that I had which had worked no issues before now no longer work.

**Comment by jackm** [comment]
Category ID **14922002** is for _Buy & Sell>Audio>Speakers_
Category ID **14922001** is for _Buy & Sell>Audio>Stereo Systems & Home Theatre_

In either case, both categories do not have any extra ad parameters needed aside from the usual "_forsaleby_" parameter, i.e. `postAdForm.attributeMap[forsaleby_s]: ownr` in the ad yaml file.

Does the html dump file say anything about what the errors are?

**Comment by Rings4** [comment]
> 
> 
> Category ID **14922002** is for _Buy & Sell>Audio>Speakers_
> Category ID **14922001** is for _Buy & Sell>Audio>Stereo Systems & Home Theatre_
> 
> In either case, both categories do not have any extra ad parameters needed aside from the usual "_forsaleby_" parameter, i.e. `postAdForm.attributeMap[forsaleby_s]: ownr` in the ad yaml file.
> 
> Does the html dump file say anything about what the errors are?

Ah, I was indifferent with which category it used, so both worked either way. I wanted Speakers, but when posting manually on kijiji I chose the other one since my item falls under both categories.

The dump says "Please fix the errors on the page" but doesn't show any other errors, and the html page is like an incompletely formed ad post page.

**Comment by jackm** [comment]
Are you sure that the ad title and ad description are long enough? The ad title length is 8 to 64 characters and the minimum ad description length is 10 characters.

**Comment by Rings4** [comment]
> Are you sure that the ad title and ad description are long enough? The ad title length is 8 to 64 characters and the minimum ad description length is 10 characters.

Yes, I'm sure. Remember that this happens for the yml files I've been using for months now as well, which would not have worked if those conditions weren't met.
